include/fastdds/dds/publisher/DataWriter.hpp

```diff
@@ -113,73 +109,86 @@ class DataWriter : public DomainEntity
      * Write data with handle.
      * @param data Pointer to the data
      * @param handle InstanceHandle_t.
-     * @return True if correct
-     * @par Calling example:
-     * @snippet fastrtps_example.cpp ex_PublisherWrite
+     * @return RETCODE_PRECONDITION_NOT_MET if the handle introduced does not match with the one associated to the data,
+     * RETCODE_OK if the data is correctly sent and RETCODE_ERROR otherwise.
      */
     RTPS_DllAPI ReturnCode_t write(
             void* data,
             const fastrtps::rtps::InstanceHandle_t& handle);
 
     /**
      * Returns the DataWriter's GUID
+     * @return Reference to the DataWriter GUID
      */
     RTPS_DllAPI const fastrtps::rtps::GUID_t& guid();
 
     /**
      * Returns the DataWriter's InstanceHandle
+     * @return Copy of the DataWriter InstanceHandle
      */
     RTPS_DllAPI fastrtps::rtps::InstanceHandle_t get_instance_handle() const;
 
     /**
-     * Get topic data type
-     * @return Topic data type
+     * Get data type associated to the DataWriter
+     * @return Copy of the TypeSupport
      */
     RTPS_DllAPI TypeSupport get_type() const;
 
     /**
      * Waits the current thread until all writers have received their acknowledgments.
+     * @param max_wait Maximum blocking time for this operation
+     * @return RETCODE_OK if the DataWriter receive the acknowledgments before the time expires and RETCODE_ERROR otherwise
```

Comment:
What about RETCODE_TIMEOUT?

Reply:
Even if the DDS Standard states that this method should return RETCODE_TIMEOUT, that situation is not differentiated by now.